ORION/scripts/analyze-batch.py

```
@@ -0,0 +1,696 @@
+
+    def _discover_fingerprint_fields(self) -> List[str]:
+        """Discover fingerprint fields from ES index mapping.
+
+        Queries the mapping, subtracts NON_FINGERPRINT_FIELDS, returns sorted list.
+        Hard fails if ES is unreachable — no fallback.
+        """
+        import requests
+
+        url = f"{self.es_server}/{self.es_index}/_mapping"
+        try:
+            response = requests.get(url)
```

Ruff (0.15.21)

[error] Probable use of `requests` call without timeout

(S113)